### PR TITLE
feat: support vite option

### PR DESCRIPTION
**Pull request by antfu**
Working example: https://github.com/antfu/vitesse-nuxt (via patch-package)

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -4,7 +4,12 @@ import { createVuePlugin } from 'vite-plugin-vue2'
 import type { ViteBuildContext } from './vite'
 
 export async function buildClient (ctx: ViteBuildContext) {
-  const clientConfig: vite.InlineConfig = vite.mergeConfig(ctx.config, {
+  const localConfig = await vite.loadConfigFromFile({
```
Comment (pi0) [inline src/client.ts:7]: Can we load it once in `src/vite.ts`?

**Comment by pi0** [comment]
Thanks for PR @antfu. It looks a good feature to have but honestly i finally prefer to use `vite` in `nuxt.config` to provide nuxt options. This way we don't fragment nuxt ecosystem (config support) and modules can use same config to extend/support vite...

**Comment by antfu** [comment]
@pi0 Sure thing. I will do that. Meanwhile, do you have any tips to extend the config typescript declarations for the `vite` entry?

**Comment by pi0** [comment]
- [config](https://github.com/nuxt/components/blob/f396cfa401986a210c118e11d142ed87c932d4ff/src/types.ts#L43)
- [hooks](https://github.com/nuxt/components/blob/main/src/types.ts#L50)

**Comment by pi0** [comment]
Thanks! Do you mind adding type extends as well?

**Comment by antfu** [comment]
Haha, just added. Am I doing right? The build somehow does not work for me:

image

And the dist does not seem to contain `.d.ts` file. I am not familiar with `siroc`, is there anything I need to setup?

**Comment by pi0** [comment]
For siroc, we only need to specify `types` field in package.json and rest is magic 🧙

**Comment by antfu** [comment]
Thanks!

**Pull request merged**